### PR TITLE
add storybook auto deploy

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

File: .travis.yml [hunk 1]
```diff
@@ -4,3 +4,7 @@ node_js:
   - '10'
   - '12'
 script: ./script/travis_run_js_tests.sh
+ 
+after_success:
+  - chmod ugo+x ./script/storybook-deploy.sh
```
Comment (sharvit) [inline .travis.yml]: I don't think it is needed, you can run `chmod +x ./script/storybook-deploy.sh` in your local env once and then push it to github.

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

File: script/storybook-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    DEPLOY_SUBDOMAIN=`echo "$DEPLOY_SUBDOMAIN_UNFORMATTED" | sed -r 's/[\/|\.]+/\-/g'`
+    DEPLOY_DOMAIN=https://${DEPLOY_SUBDOMAIN}-${REPO_NAME}-${REPO_OWNER}.surge.sh
+    surge --project ${DEPLOY_PATH} --domain $DEPLOY_DOMAIN;
+    if [ "$TRAVIS_PULL_REQUEST" != "false" ]
+    then
+      # Using the Issues api instead of the PR api
+      # Done so because every PR is an issue, and the issues api allows to post general comments,
+      # while the PR api requires that comments are made to specific files and specific commits
+      GITHUB_PR_COMMENTS=https://api.github.com/repos/${TRAVIS_REPO_SLUG}/issues/${TRAVIS_PULL_REQUEST}/comments
+      curl -H "Authorization: token ${GITHUB_API_TOKEN}" --request POST ${GITHUB_PR_COMMENTS} --data '{"body":"Travis automatic deployment: '${DEPLOY_DOMAIN}'"}'
```
Comment (sharvit) [inline script/storybook-deploy.sh:29]: nitpik: 'Storybook automatic deployment'

**Comment by amirfefer** [comment]
Travis automatic deployment: https://14-pr-foreman-amirfefer.surge.sh

**Comment by amirfefer** [comment]
I'm closing this because Travis won't let secret env vars to be read on pull requests

**Pull request closed without merging**